src/Agent.Worker/JobExtension.cs

```diff
@@ -726,6 +727,32 @@ private void OutputSetupInfo(IExecutionContext context)
             }
         }
 
+        private void PublishKnobsInfo(IExecutionContext jobContext)
+        {
+            string jobId = jobContext?.Variables?.System_JobId?.ToString() ?? string.Empty;
+
+            var telemetryData = new Dictionary<string, string>()
+            {
+                { "JobId", jobId }
+            };
+
+            foreach (var knob in Knob.GetAllKnobsFor<AgentKnobs>())
+            {
+                var value = knob.GetValue(jobContext);
+                if (value.Source.GetType() != typeof(BuiltInDefaultKnobSource))
+                {
+                    var stringValue = value.AsString();
+                    if (knob is SecretKnob)
+                    {
+                        HostContext.SecretMasker.AddValue(stringValue, $"JobExtension_InitializeJob_{knob.Name}");
+                    }
```

Comment:
Isn't this made already?
https://github.com/microsoft/azure-pipelines-agent/blob/f60062e975aab1c9b0a4bc9ec4a87b3e61c081e5/src/Agent.Worker/JobExtension.cs#L186-L189